libr/util/strbuf.c

```diff
@@ -12,39 +12,52 @@ R_API RStrBuf *r_strbuf_new(const char *str) {
 	return s;
 }
 
+R_API bool r_strbuf_equals(RStrBuf *sa, RStrBuf *sb) {
+	if (!sa || !sb || sa->len != sb->len) { // faster comparisons
+		return false;
+	}
+	return strcmp (r_strbuf_get (sa), r_strbuf_get (sb)) == 0;
+}
+
+R_API int r_strbuf_length(RStrBuf *sb) {
+	return sb? sb->len: 0;
+}
+
 R_API void r_strbuf_init(RStrBuf *sb) {
 	memset (sb, 0, sizeof (RStrBuf));
 }
 
-R_API bool r_strbuf_set(RStrBuf *sb, const char *s) {
-	int l;
-	if (!sb) {
-		return false;
-	}
-	if (!s) {
-		r_strbuf_init (sb);
-		return true;
-	}
-	l = strlen (s);
+R_API bool r_strbuf_setbin(RStrBuf *sb, const ut8 *s, int l) {
 	if (l >= sizeof (sb->buf)) {
```

Comment:
Should `		R_FREE (sb->ptr);` be called here?

I am not clear know you use `r_strbuf_set`, but if you allocate with `malloc ( l + 1)`, shouldn't it be NUL terminated?

Reply:
not necesarily. if its used as 'bin' it is not